extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/security/AuthorizationPolicy.java

```diff
@@ -0,0 +1,27 @@
+package io.quarkus.vertx.http.security;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import io.quarkus.vertx.http.runtime.security.HttpSecurityPolicy;
+
+/**
+ * Secures endpoint classes and methods with {@link HttpSecurityPolicy}.
+ * Policies selected by this annotation will run right after all path-matching policies.
```

Comment:
Just by reading this I am not sure I would understand how much it can do for me. Maybe it would make sense to have some kind example in the Javadocs?

Reply:
You are right, that Javadoc doesn't explain a thing unless you know HttpSecurityPolicy. This policy can both authorize request (access identity, routing context) and augment identity (which I don't plan to document for now as augmentors are the way to go). I'll add example of HttpSecurityPolicy here in a moment.

Reply:
🙏🏽